### PR TITLE
Fix for overflowing api key hint text on certain zoom levels.

### PR DESCRIPTION
**Pull request by agr**
Addresses #8190 

It seems that at some point, in order to align the bottoms of the package list and the API key hint container their respective heights were explicitly specified in CSS (see diff). At certain zoom levels (and browser window widths) the contents of the hint container gets longer than the container permits and overflows:
image

In order to get rid of hardcoding container heights a bunch of flex containers were introduced, hint container was wrapped with an additional `<div class="form-group">` in order to layout the contents in the same way as in the container on the left (i.e. to have the same margin-bottom), so bottom border end up on the same level.
At the same time, table inside the hint container was styled to not have a bottom margin to eliminate weird white space:
image

Resulting in the following look (at various widths and zoom levels):

image

image

image

### CODE REVIEW

File: src/Bootstrap/less/theme/page-api-keys.less [hunk 1]
```diff
@@ -42,15 +42,36 @@
     .available-packages {
       .panel-body {
         overflow: auto;
-        height: 204px;
+        min-height: 180px;
```
Comment (agr) [inline src/Bootstrap/less/theme/page-api-keys.less:45]: This is set to be a little bit less than the height of a hint container on any zoom level, so it only takes effect when all is laid out in a single column.
Should I add a media query around it to enforce the behavior?

Comment (agr) [reply src/Bootstrap/less/theme/page-api-keys.less]: Added media queries.

**Review by joelverhagen — APPROVED** [review]
Wow, great improvement!

**Review by loic-sharma — APPROVED** [review]

**Review by dannyjdev — APPROVED** [review]

**Comment by agr** [comment]
This solution misbehaves when the package list is large (it stretches the container instead of adding a scroll bar), looking for a fix.

File: src/NuGetGallery/Views/Users/ApiKeys.cshtml [hunk 2]
```diff
@@ -415,59 +415,63 @@
...
         </div>
         <div class="row select-packages-form" data-bind="css: { disabled: !SelectPackagesEnabled() }">
-            <div class="col-sm-7">
-                <div class="form-group">
-                    <input type="text" class="form-control" data-val="true"
-                           data-val-regex="Invalid glob pattern. See the example glob patterns below for further details."
-                           data-val-regex-pattern="^[\*\w_.-]*$"
-                           data-bind="textInput: PendingGlobPattern,
+            <div class="col-sm-7 select-packages-inputs">
+                <div>
```
Comment (agr) [inline src/NuGetGallery/Views/Users/ApiKeys.cshtml:419]: New div added here to allow absolute positioning.

**Comment by agr** [comment]
I think I found a [hacky] solution for long lists: added a wrapper around the left section with `position: absolute` to prevent the content of that div to contribute to the size of the container (but still positioned to occupy all available vertical space), so now the height of `.select-packages-inputs` element is determined by the only remaining item - API hint section.
All the container elements down to the list of packages were made flex-boxes, so they would stretch to occupy the available vertical space, an `overflow: auto` was moved one level above to reduce the amount of nested flexboxes.

**Pull request merged**